### ISSUE TITLE
Mlflow Mixin

### ISSUE THREAD

**Issue by sailngarbwm** [rbturnbull/fastapp#1]
The task is to write an mlflow mixin that is similar to the wandbmixin. See here for some more details: https://rbturnbull.github.io/fastapp/logging.htmlhttps://rbturnbull.github.io/fastapp/logging.html

The necessary functions:

- init_run - this will start an mlflow run, with a given experiment, and optional run name. If no experiment_id is given, then it should create, or get the experiment with the same name as the app class (note that self.project_name() should give you that by default). One other thing is that this should initiate `mlflow.fastai.autolog(log_models=False)`. There should also be an optional `tracking_uri' that can be set using mlflow.set_tracking_uri

- log - this function should take in a dictionary, and log it as a set of parameters, or if the parameter metric = True, then it logs things as a set of metrics, with an optional argument 'step' that goes into the log_metrics function

- log_artifact - This is a bit different from wandb, as wandb requires you to save the file to disk before you can log it, what I would like is for you to be able to input an `artifact` that could be a pandas dataframe, maltplotlib or plotly.figure, dictionary, string, or path to a saved file. As well as a string or path that declares the path to the artifact when saved in the artifact store (ie 'model_results/outputs.csv')
   - Note there are some functions to help with this in mlflow like `mlflow.log_text`, `mlflow.log_dict`, `mlflow.log_figure`, that will help with some things, other things like a pandas dataframe, you should save to a temporary csv, and log that saved file, Any other type that isn't mentioned above, use pickle to save the object, and log that as an artifact.

- save_model - for this function, I would like to save the model as a pytorch model. This function will take in a fastai.learner so and you can access the pytorch version of the model from a learner.model, and use the `mlflow.pytorch.log_model` function
